runtime/v2/README.md

```diff
@@ -1,29 +1,177 @@
+
+This document describes the major components of the v2 runtime integration model, how the components interact
+with containerd and the v2 runtime, and how to use and integrate different v2 runtimes.
+
+To simplify the interaction, runtime v2 introduced a first class v2 API for runtime authors to integrate with containerd,
+replacing the v1 api.
+The v2 API is minimal and scoped to the execution lifecycle of a container.
+
+This document is split into the following sections:
+
+* [architecture](#architecture) - the major components, their purposes and relationships
+* [usage](#usage) - how to invoke specific runtimes, and how to configure them
+* [authoring](#shim-authoring) - how to author a v2 runtime
+
+## Architecture
+
+### containerd-runtime communication
+
+containerd expects a runtime to implement several container control features, such as create, start and stop.
+The most common runtime used is [runc](https://github.com/opencontainers/runc).
+
+The high-level flow is as follows:
+
+1. client requests from containerd to create a container
+1. containerd lays out the container's filesystem, and creates the necessary config information
+1. containerd invokes the runtime, such as `runc`, to create/start/stop the container
+
+However, containerd does not actually directly invoke the runtime to start the container.
+Instead it expects to invoke the runtime, which will expose a socket - Unix-domain on Unix-like systems, named pipe on Windows -
+and listen for container commands via [ttRPC](https://github.com/containerd/ttrpc) over that 
+socket.
+
+The runtime is expected to process those operations. How it does so is entirely within the scope of the runtime implementation.
+Two common patterns are:
+
+* a single binary for runtime that both listens on the socket and creates/starts/stops the container
+* a separate shim binary that listens on the socket, and invokes a separate runtime engine that creates/starts/stops the container
+
+The separate "shim+engine" pattern is used because it makes it easier to integrate distinct runtimes implementing the OCI
+runtime spec. The ttRPC protocol can be handle via one runtime shim, while distinct runtime engine implementations can
+be used, as long as they implement the OCI runtime spec.
+
+#### Architecture
+
+##### runtime shim
+
+The runtime shim is what actually is invoked by containerd. It has minimal options on start beyond
+being provided the communications port for containerd and some configuration information.
+
+The runtime shim listens on the socket for ttRPC commands from containerd, and then invokes a separate actual OCI runtime
+engine via `fork`/`exec`.
+
+containerd passes options to the shim over the ttRPC connection, including which runtime binary
+to invoke. These are the `options` for the [`CreateTaskRequest`](https://github.com/containerd/containerd/blob/main/runtime/v2/README.md#container-level-shim-configuration).
+
+##### runtime engine
```

Comment:
```suggestion
##### runtime engine (OCI runc or compatible) 
```

Reply:
Is that correct? My understanding (now that I understand it), is that the only part set by containerd is the ttRPC API, i.e. that containerd expects to:

1. Start a process
2. Process exits and returns a socket address
3. That socket has a ttRPC listener available

The split between shim and engine is internal to the runtime implementation. It need not by OCI at all, it could be not split, etc. 

Granted the `runc-v2` shim that ships with containerd _does_ specifically shim to an OCI compatible runtime engine, but that is a specific implementation.

Reply:
yeah that was my point.. you can have a runc compatible or some other runtime